packages/contracts/src/interface/ICredbull.sol

```diff
@@ -27,6 +27,56 @@ interface ICredbull {
         uint256 depositThresholdForWhitelisting;
     }
 
+    struct BaseVaultParams {
+        IERC20 asset;
+        string shareName;
+        string shareSymbol;
+        address custodian;
+    }
+
+    struct ContractRoles {
+        address owner;
+        address operator;
+        address custodian;
+    }
+
+    struct WindowParams {
+        uint256 opensAt;
+        uint256 closesAt;
+    }
+
+    struct WindowVaultParams {
+        WindowParams depositWindow;
+        WindowParams matureWindow;
+    }
+
+    struct KycParams {
+        address kycProvider;
+        uint256 depositThresholdForWhitelisting;
+    }
+
+    struct MaxCapParams {
+        uint256 maxCap;
+    }
+
+    struct FixedYieldVaultParams {
```

Comment:
Are we not moving the specific param Structs to the associated contract?

Reply:
As discussed, this is already a large PR, so moving the 'params' structs will be another PR.